### ISSUE TITLE
 Duplicated field names in protobuf messages causes wrong request transformation

### ISSUE THREAD

**Issue by kgoguevgoget** [ysugimoto/lua-resty-grpc-gateway#14]
The Lua request table after transformation is wrong if the same protobuf field name is used in repeated message types

given the following basic example. This is perfectly fine in proto to do this
```
syntax = "proto3";

package helloworld;

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
}

message HelloRequest {
  string name = 1; /** name declared here first**/
  repeated Locations locations =2;
}

message Locations {
  string name =1; /**second declaration of name **/
}

message HelloReply {
  string message = 1;
}
```
Input:
`curl -vv -H "Content-Type: application/json" -d '{"name":"grpc-rest","locations":[{"name":"test123"}]}' "http://localhost:9000/rest"`

Expected Lua output:

```
  request: {
   locations = { {
       name = "test123"
     } },
   name = "grpc-rest"
 }
```
However due to the way that we setup the internal Lua request table The following is the output

```
  request: {
   locations = { {
       name = "grpc-rest"
     } },
   name = "grpc-rest"
 }
```

The bug occurs because of the way we now handle the default_values and the function `get_from_request` if there are repeated user defined message types. 

Based on the current code, for each protobuf type we perform the following:
 
1. Check if the field type starts with a '.' - denoting that is a user defined type
2. Ensure that we setup the `default_values` table  once with everything that is coming in from the OpenResty request
3. If the type of the protobuf field has the repeated label then iterate over those elements and eventually build the request table as normal.
4. For each key/value pair we call `get_from_request` which eventually checks and sets the type of the data for building the request table. 

This generally works very well as long as the keys of the Lua table are differently named. 

If we have the keys in the request table which are named the same, as shown in the basic example, then when we return from the recursion  the output of the following will cause this bug. 

https://github.com/ysugimoto/lua-resty-grpc-gateway/blob/a53ac1edf1fd216b71fd222df1d7f76efbbdb080/grpc-gateway/util.lua#L109

`request[locations] = name="grpc-rest" or name="test123" or nil`

The function `get_from_request(name,kind)` will return `name="grpc-rest` due to the `request_table` getting filled from the ngx GET/POST, Since the key that we are looking for already exists, we end up returning that one.

Thus when we perform  the following with OR statements.  It is perfectly normal that the first value (grpc-rest) would get picked Instead of the correct one (test123)

`request[locations] = name="grpc-rest" or name="test123" or nil`

A possible solution would be to do the following:
1. Rewrite `get_from_request` - It should not constantly check the ngx.req, instead only concern itself with setting the appropriate type for the data. Naturly this will have to take in the default_values[name] as we need to keep track of the object we are working with
https://github.com/ysugimoto/lua-resty-grpc-gateway/blob/a53ac1edf1fd216b71fd222df1d7f76efbbdb080/grpc-gateway/util.lua#L42-L50
2. Remove the following from inside the `map_message` and do this outside, only once in the lifecycle of the openResty request, then call `map_message` with the `default_values` always populated
https://github.com/ysugimoto/lua-resty-grpc-gateway/blob/a53ac1edf1fd216b71fd222df1d7f76efbbdb080/grpc-gateway/util.lua#L77-L86

**Comment by ysugimoto** [comment]
@kgoguevgoget I totally agree with you. the `map_message` seems a little bit fragile and should rewrite as well.

**Comment by kgoguevgoget** [comment]
@ysugimoto I might be able to get this working, depending on my work. I will see if I have some time later to put up a PR. It might or might not be pretty 😅

**Comment by kgoguevgoget** [comment]
@ysugimoto I might be able to get this working, depending on my work. I will see if I have some time later to put up a PR. It might or might not be pretty 😅

**Comment by kgoguevgoget** [comment]
Hi, just wanted to say I haven't forgotten about this, most likely next week I will get some time to take a look at this

**Comment by kgoguevgoget** [comment]
@ysugimoto I managed to get this to work. I am just doing some quick testing then I shall put up a PR for this later today hopefully.  :)

**Comment by kgoguevgoget** [comment]
@ysugimoto This should fix the issue properly. As per our discussions above the solution has been implemented. Added a test to check this, also checked it by manually running a few tests through. From my end its doing the correct thing now :)

**Comment by kgoguevgoget** [comment]
Latest Release now fixes this issue merged with PR #15
